File: Python/src/hardware/controller.py
import wpilib
from wpilib.joystick import Joystick

import math

class Controller(Joystick):

    def _sgn(val):
        if val > 0:
            return 1
        elif val == 0:
            return 0
        else:
            return -1

    def _pow_scale(val, pw=2.0):
        _abs_v = abs(val)
        _sgn_v = Controller._sgn(val)
        return _sgn_v * (_abs_v ** pw)

    def _exp_scale(val, base=2**5):
        _abs_v = abs(val)
        _sgn_v = Controller._sgn(val)
        return _sgn_v * ((base**_abs_v-1)/(base-1))

    def getAxis(self, id):
        #return self.getRawAxis(id)
        return Controller._pow_scale(self.getRawAxis(id))

File: src-java/src/main/java/frc3966/robot/hardware/Controller.java
package frc3966.robot.hardware;


import edu.wpi.first.wpilibj.Joystick;

/**
 * PS4 controller wrapper for a joystick.
 * @author Matthew.Lythgoe
 */
public class Controller extends Joystick {
   
    
    public Controller(int port) {
        super(port);
    }

    private double _pow_scale(double val, double pow) {
        double _abs_v = Math.abs(val);
        double _sgn_v = Math.signum(val);
        return _sgn_v * Math.pow(_abs_v, pow);
    }

    private double _pow_scale(double val) {
        return _pow_scale(val, 2.0);
    }
    
    public boolean getButton(int port) {
        return getRawButton(port);
    }

    public double getAxis(int port) {
        return _pow_scale(getRawAxis(port));
    }
}
